index.js

```diff
@@ -1,28 +1,65 @@
 const readline = require('readline');
 const readlineInterface = readline.createInterface(process.stdin, process.stdout);
 
+
+
+function outsideRoom (answer) {
+if (answer === 'read sign'){
+  return 'The sign says "Welcome to Burlington Code Academy! Come on up to the third floor. If the door is locked, use the code 12345."'
+}else if (answer === 'gargle'){
+  return "Sorry, I don't know how to gargle."
+}else if(answer === 'take sign'){
+  return 'That would be selfish. How will other students find their way?'
+}else if(answer === 'enter code 12345' || answer === 'key in 12345'){
+  console.log('Success! The door opens. You enter the foyer and the door shuts behind you.')
+  return enterRoom(foyer)
+}else if(answer === 'enter code 54321'){
+  return 'Bzzzzt! The door is still locked.'
+}else {
+  console.log('The door is locked. There is a keypad on the door handle.')
+}
+}
+
+function foyer (answer) {
+  console.log(answer + 'in the foyer')
+}
+
+function mrMikes (answer) {
+  console.log(answer + 'I am at mr.Mikes ')
+}
+
+function muddyWaters (answer) {
+  console.log(answer + 'I am at muddy waters')
+}
+
+function classroom (answer) {
+  console.log(answer + 'I am learning!')
+}
+
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
 
 // remember the StateMachine lecture
-// https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
+// https://bootcamp.burlingtoncodeacademy.com/lessons/javascript/state-machines
+let rooms = {
+  'outside': { canChangeTo: [foyer, mrMikes, muddyWaters] },
+  'foyer': { canChangeTo: [classroom, outside] },
+  'classroom': { canChangeTo: [foyer] },
+  'muddyWaters': { canChangeTo: [outside] },
+  'mrMikes': {canChangeTo: [outside]}
```

Comment:
Your room names are strings, so you should pass them into your state machine as strings e.g. `canChangeTo: ['foyer', 'mrMikes', 'muddyWaters']` otherwise you'll get an error because the variable has not yet been defined